src/cookbook/work-with-non-mintable-assets.md

```diff
@@ -11,4 +11,48 @@ head:
 
 # How to Work with Non-Mintable Assets
 
-TODO
\ No newline at end of file
+```rust
+fn register_non_mintable_asset(
+    iroha: &Client,
+) {
+    let magical_keys = AssetDefinitionId::from_str("magical_key#wonderland").unwrap();
+    // register keys as an asset definition
+    let register_keys_as_a_concept = Register::asset_definition(
+        AssetDefinition::new(
+            magical_keys.clone(),
+            AssetValueType::Numeric(NumericSpec::integer()),
+        ).mintable_once()
+    );
+    let alice = "alice@wonderland".parse().unwrap();
+    // Alice owns ten keys and cannot mint more
+    let initial_keys_of_alice = Asset::new(
+        AssetId::new(roses, alice), 
+        10_u32
+    );
+    let register_keys_of_alice = Register::asset(initial_keys_of_alice);
+    iroha.submit_all([
+        InstructionBox::from(register_keys_as_a_concept),
+        InstructionBox::from(register_keys_of_alice)
+    ]).unwrap();
+}
+```
+
+```rust
+fn mint_non_mintable_asset(
```

Comment:
If it's non-mintable, how it could be minted 🤔

Reply:
Good question, I think the mintability naming is unfortunate, and we should rethink it. But that's an issue for the main repo

Reply:
Same as https://github.com/hyperledger/iroha-2-docs/pull/424#discussion_r1607990016